### DISCUSSION TITLE
How to have a commend to set tool digital output not digital output

### DISCUSSION THREAD

**Discussion by WiesenDX** [visose/Robots#222] [Q&A] [answered]
I'm trying to control a gripper on UR10e, the gripper only read **tool digital output**0,1. But here we can only have a **digital output** battery. Anyone know how to deal with it?[图片]

**Comment by WiesenDX** [comment]
![微信图片_20230314181425](https://user-images.githubusercontent.com/127881362/225085215-ca47aa48-6f50-4169-b0a7-74eb8d0b590f.jpg)

**Reply by visose** [reply]
You can use the custom command component with: `set_tool_digital_out(1, True)`
(you can use 0 or 1, and True or False)

**Reply by WiesenDX** [reply]
Yes, I have done in this way, it works well. Thank you!

**Comment by visose** [comment, marked as answer]
For UR tool digital outputs, use a Custom Command with URScript such as `set_tool_digital_out(0, True)` or `set_tool_digital_out(1, False)`. The built-in SetDO command targets controller digital outputs, while the UR tool output is exposed through URScript.
